src/LogBuffer.cpp

```
@@ -0,0 +1,68 @@
+        for (size_t i = 0; i < available; i++) {
+            out[i] = buffer_[(start + i) % BUFFER_SIZE];
+        }
+        out[available] = '\0';
+        xSemaphoreGive(mutex_);
+        return available;
+    }
+    out[0] = '\0';
+    return 0;
+}
+
+void LogBuffer::clear() {
```

Clang (14.0.6)

[warning] method 'clear' can be made static

(readability-convert-member-functions-to-static)